mopidy_musicbox_webclient/static/js/controls.js

```diff
@@ -604,6 +613,36 @@ function showFavourites() {
     });    
 }
 
+function upgradeStreamUrisToFavourites() {
+    $.cookie.json = true;
+    var streamUris = $.cookie('streamUris'); // Read the cookie.
+    if (streamUris) {
+        var uris = []; // Prepare a list of uris to lookup.
+        for (var key in streamUris) {
+            var rs = streamUris[key];
+            if (rs) {
+                uris.push(rs[1]);
+            }
+        }
+        mopidy.library.lookup({'uris': uris}).then(function(results) {
+            var tracks = []; // Prepare a list of tracks to add.
+            for (var key in streamUris) {
+                var rs = streamUris[key];
+                if (rs) {
+                    var track = results[rs[1]][0];
+                    track.name = rs[0] || track.name; // Use custom name if provided.
```

Comment:
No, ``lookup`` doesn't raise an exception as I originally thought, it just returns a dictionary without a value for the URI key. 

So this works as far as I can tell.